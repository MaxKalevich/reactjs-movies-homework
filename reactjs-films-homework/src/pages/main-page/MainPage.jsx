import MovieCard from "../../components/movie-card/MovieCard";
import PaginationContainer from "../../components/pagination/PaginationContainer";
import SwitchPanelContainer from "../../components/switch-panel/SwitchPanelContainer";

import { Link } from "react-router-dom";

import styles from "./main-page_style.module.scss";

const MainPage = ({
  setPage,
  topMovies,
  searchMovies,
  searchFromUrl,
  api_category,
  language,
  search,
}) => {
  const moviesData =
    topMovies !== undefined &&
    topMovies.map((movie) => {
      return (
        <Link
          className={styles.link}
          to={`movie/${movie.id}?filter=${api_category}&name=${movie.title}&lang=${language}`}
          key={movie.id}
        >
          <MovieCard
            movieTitle={movie.title}
            movieImage={movie.poster_path}
            estimation={movie.vote_average}
            date={movie.release_date}
            setPage={setPage}
            movieId={movie.id}
            key={movie.id}
          />
        </Link>
      );
    });

  const searchMoviesData =
    searchMovies !== undefined &&
    search !== undefined &&
    searchMovies.map((movie, index) => {
      return (
        <Link
          className={styles.link}
          to={`movie/${movie.id}?filter=${api_category}&name=${movie.title}&lang=${language}&search=${search}`}
          key={movie.id}
        >
          <MovieCard
            movieTitle={movie.title}
            movieImage={movie.poster_path}
            estimation={movie.vote_average}
            genre={movie.genre}
            setPage={setPage}
            key={movie.title + index}
          />
        </Link>
      );
    });

  return (
    <section className={styles.mainPage}>
      <SwitchPanelContainer />
      <main className={styles.cardContainer}>
        {searchFromUrl !== undefined &&
          searchFromUrl.length !== 0 &&
          searchMovies.length === 0 && <h2>NO RESULTS FOUND</h2>}
        {(search !== undefined && search.length) || searchFromUrl
          ? searchMoviesData
          : moviesData}
      </main>
      <PaginationContainer />
    </section>
  );
};

export default MainPage;
