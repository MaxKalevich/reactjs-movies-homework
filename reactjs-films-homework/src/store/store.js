import { combineReducers, configureStore } from "@reduxjs/toolkit";
import mainPageSlice from "./reducers/mainPageSlice";
import appSlice from "./reducers/appSlice";
import movieDetailsPageSlice from "./reducers/movieDetailsPageSlice";
import actorPageSlice from "./reducers/actorPageSlice";

const rootReducer = combineReducers({
  mainPageSlice,
  appSlice,
  movieDetailsPageSlice,
  actorPageSlice,
});

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(),
  devTools: process.env.NODE_ENV !== "production",
});
