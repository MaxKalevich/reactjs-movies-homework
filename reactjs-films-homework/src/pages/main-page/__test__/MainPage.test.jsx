import { render } from "@testing-library/react";
import MainPage from "../MainPage";
import { Provider } from "react-redux";
import { store } from "../../../store/store";

describe("Checking MainPage component", () => {
  it("Snapshot", () => {
    const mainPageMockData = [];
    const { asFragment } = render(
      <Provider store={store}>
        <MainPage mainPageMockData={mainPageMockData} />
      </Provider>
    );
    expect(asFragment(<MainPage />)).toMatchSnapshot();
  });
});
