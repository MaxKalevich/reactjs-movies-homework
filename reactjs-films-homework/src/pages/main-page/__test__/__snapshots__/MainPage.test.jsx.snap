// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking MainPage component Snapshot 1`] = `
<DocumentFragment>
  <section
    class="mainPage"
  >
    <div
      class="switchWrapper"
    >
      <ul
        class="itemList"
      >
        <li
          class="currentCategory"
        >
          Popular
        </li>
        <li
          class="category"
        >
          Top rated
        </li>
        <li
          class="category"
        >
          Upcoming
        </li>
      </ul>
    </div>
    <main
      class="cardContainer"
    />
    <div
      class="pages"
    >
      <span
        class="currentPage"
      >
        1
      </span>
      <span
        class="page"
      >
        2
      </span>
      <span
        class="page"
      >
        3
      </span>
      <span
        class="page"
      >
        4
      </span>
      <span
        class="page"
      >
        5
      </span>
    </div>
  </section>
</DocumentFragment>
`;
